### PR TITLE
feat(controller): support proxying to the serving model workload

### PR DESCRIPTION
**Pull request by jialeicui**
## Description

live demo

```bash
# curl -s http://jialei.pre.intra.starwhale.ai/gateway/model-serving/20/ppl -X POST -F "data=@/home/foo/foo" | jq
[
  [
    9
  ],
  [
    [
      3.3270520369869727e-09,
      1.2171811490062852e-10,
      1.2030677147728627e-09,
      1.7674929805337285e-09,
      9.05094143771982e-05,
      6.260217371438276e-08,
      4.5112372466471804e-10,
      4.308098875417157e-06,
      1.1480706309613537e-05,
      0.999893676619871
    ]
  ]
]
```

api spec in model
http://jialei.pre.intra.starwhale.ai/gateway/model-serving/20/api-spec

## Modules
- [ ] UI
- [x] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: server/controller/src/main/java/ai/starwhale/mlops/common/ProxyServlet.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+        var host = new HttpHost(URIUtils.extractHost(uri));
+        var request = generateRequest(req, uri.getPath());
+        var response = httpClient.execute(host, request);
+        generateResponse(response, res);
+    }
+
+    protected HttpRequest generateRequest(HttpServletRequest req, String uri) throws IOException {
+        var request = new BasicHttpEntityEnclosingRequest(req.getMethod(), uri);
+        request.setEntity(new InputStreamEntity(req.getInputStream(), req.getContentLength()));
```
[the remark this answers is not in the document]
Comment (jialeicui) [reply server/controller/src/main/java/ai/starwhale/mlops/common/ProxyServlet.java:73]: - Model serving does not support params registration in the `@api` decorator for now 
- I think the params can be handled correctly in `getTarget` func

[the remark this answers is not in the document]
Comment (anda-ren) [reply server/controller/src/main/java/ai/starwhale/mlops/common/ProxyServlet.java:73]: > * Model serving does not support params registration in the `@api` decorator for now

OK
> * I think the params can be handled correctly in `getTarget` func


`var request = generateRequest(req, uri.getPath());` still strips the params

**Review by anda-ren — APPROVED** [review]
LGTM

File: server/controller/src/main/java/ai/starwhale/mlops/common/ProxyServlet.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jialeicui) [reply server/controller/src/main/java/ai/starwhale/mlops/common/ProxyServlet.java:78]: Yes, you're right, adds the query string from URL to the request. (posted form urlencoded not support for now, it's a little complicated)

**Review by anda-ren — APPROVED** [review]
LGTM

**Pull request merged**